taskweaver/role/translator.py

```diff
@@ -262,66 +256,34 @@ def parse_llm_output_stream_v2(
-                        cur_type, cur_content = None, None
-                    if ev.event == "end_map":
-                        if cur_type is None or cur_content is None:
-                            raise Exception(
-                                f"Incomplete generate kv pair in index {item_idx}. "
-                                f"type: {cur_type} content {cur_content}",
-                            )
-
-                        if cur_content_sent and not cur_content_sent_end:
-                            # possible incomplete string, trigger end prematurely
-                            yield cur_type, "", True
-
-                        if not cur_content_sent:
-                            yield cur_type, cur_content, True
-
-                        cur_content_sent, cur_content_sent_end = False, False
-                        cur_type, cur_content = None, None
-                        item_idx += 1
-
-                if ev.prefix == cur_item_prefix + ".type":
-                    if ev.event == "string" and ev.is_end:
-                        cur_type = ev.value
-
-                if ev.prefix == cur_item_prefix + ".content":
-                    if ev.event == "string":
-                        if cur_type is not None:
-                            cur_content_sent = True
-                            yield cur_type, ev.value_str, ev.is_end
-
-                            assert not cur_content_sent_end, "Invalid state: already sent is_end marker"
-                            if ev.is_end:
-                                cur_content_sent_end = True
-                        if ev.is_end:
-                            cur_content = ev.value
+                if ev.prefix == root_element_prefix and ev.event == "map_key" and ev.is_end:
+                    cur_type = ev.value
+                    yield cur_type, "", False
+                elif ev.prefix == f"{root_element_prefix}.{cur_type}" and ev.event == "string":
+                    yield cur_type, ev.value_str, ev.is_end
+                elif ev.prefix == f"{root_element_prefix}.{cur_type}" and ev.event == "number":
+                    yield cur_type, ev.value_str, ev.is_end
+                elif ev.prefix == f"{root_element_prefix}.{cur_type}" and ev.event == "boolean":
+                    yield cur_type, ev.value_str, ev.is_end
+                elif ev.prefix == f"{root_element_prefix}.{cur_type}" and ev.event == "start_map":
+                    self.logger.warning(f"Start map in property: {root_element_prefix}.{cur_type}")
+                elif ev.prefix == f"{root_element_prefix}.{cur_type}" and ev.event == "end_map":
+                    yield cur_type, ev.value, True
+                elif ev.prefix == f"{root_element_prefix}.{cur_type}" and ev.event == "start_array":
+                    self.logger.warning(f"Start array in property: {root_element_prefix}.{cur_type}")
+                elif ev.prefix == f"{root_element_prefix}.{cur_type}" and ev.event == "end_array":
```

Comment:
for the array/object value, maybe we need to convert them to string type with `json.dumps` to match the data type expectation?

Reply:
Yes, will do